Engines/Wine/QuickScript/Zip Script/script.js

```diff
@@ -1,70 +1,76 @@
-    this._checksum = checksum;
-    return this;
-};
-
-ZipScript.prototype.go = function () {
-    var setupWizard = SetupWizard(InstallationType.APPS, this._name, this.miniature());
+include("engines.wine.verbs.luna");
 
-    setupWizard.presentation(this._name, this._editor, this._applicationHomepage, this._author);
+module.default = class ZipScript extends QuickScript {
+    constructor() {
+        super();
+    }
 
-    var wine = new Wine()
-        .wizard(setupWizard)
-        .prefix(this._name, this._wineDistribution, this._wineArchitecture, this._wineVersion)
-        .create()
-        .luna()
-        .wait();
+    url(url) {
+        this._url = url;
+        return this;
+    }
+    setupPathInsidePrefix(setupPath) {
+		this._setupPath = setupPath;
+		return this;
+	}
+
+    checksum(checksum) {
+        this._checksum = checksum;
+        return this;
+    }
 
-    this._preInstall(wine, setupWizard);
+    go() {
+        const setupWizard = SetupWizard(InstallationType.APPS, this._name, this.miniature());
 
-    // back to generic wait (might have been changed in preInstall)
-    setupWizard.wait(tr("Please wait..."));
+        setupWizard.presentation(this._name, this._editor, this._applicationHomepage, this._author);
 
-    var archive = "";
-    if (!this._url) {
-        archive = setupWizard.browse(tr("Please select the .zip file."), wine.prefixDirectory(), ["zip"]);
-    } else {
-        archive = wine.prefixDirectory() + "/drive_c/archive.zip";
-        new Downloader()
+        const wine = new Wine()
             .wizard(setupWizard)
-            .url(this._url)
-            .checksum(this._checksum)
-            .to(archive)
-            .get();
-    }
-
-    new Extractor()
-        .wizard(setupWizard)
-        .archive(archive)
-        .to(wine.prefixDirectory() + "/drive_c/" + this._name)
-        .extract();
-
-    this._postInstall(wine, setupWizard);
+            .prefix(this._name, this._wineDistribution, this._wineArchitecture, this._wineVersion)
+            .create()
+            .luna()
+            .wait();
+
+        this._preInstall(wine, setupWizard);
+
+        // back to generic wait (might have been changed in preInstall)
+        setupWizard.wait(tr("Please wait..."));
+
+        let archive = "";
+        if (!this._url) {
+            archive = setupWizard.browse(tr("Please select the .zip file."), wine.prefixDirectory(), ["zip"]);
+        } else {
+            archive = wine.prefixDirectory() + "/drive_c/archive.zip";
+            new Downloader()
+                .wizard(setupWizard)
+                .url(this._url)
+                .checksum(this._checksum)
+                .to(archive)
+                .get();
+        }
+
+        new Extractor()
+            .wizard(setupWizard)
+            .archive(archive)
+            .to(wine.prefixDirectory() + "/drive_c/" + this._name)
+            .extract();
+        if (!fileExists(this._setupPath)){
```

Comment:
The problem now: if you not provide a _setupPath (like in regular ZipScript, it will crash. I suggest:
```js
if (this._setupPath !== `undefined`){
    if (!fileExists(this._setupPath)){
        throw "Hello There"
    }
}